src/actions/index.js

```diff
@@ -55,10 +57,10 @@ export function saveCurrentProject(state) {
   return false;
 }
 
-function validateSource(language, source, enabledLibraries) {
+function validateSource(language, source, analyzer) {
```

Comment:
@alexpelan pretty much. To clarify, my proposal was that *in the context of validations*, we think of this as just being a collection of `projectAttributes`, with no particular knowledge/opinion about what is providing that collection of attributes. In contexts where we are actually constructing an analyzer instance (e.g. [here](https://github.com/popcodeorg/popcode/pull/619/files/80d41631fdc3137bad018bb4f6cae9e39bd3b002#diff-4e3037116dbc2bc69a473b7ace3d02abR84)), I think it still makes sense to call the variable `analyzer`.

However my intent definitely isn’t for you to make changes you don’t agree with / don’t make sense. Please let me know if that is the case here!